hazelcast/src/main/java/com/hazelcast/internal/metrics/StringGauge.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
```

Comment:
There is no object creation after start for the gauge. It is initialized and reused, hence it should not cause any litter. The gauge is cached in Statistics.PeriodicStatistics object. 

The reason why defined the string gauge is this: in construction at PeriodicStatistics I only know the name of the gauge but do not know if the type is long or double, hence StringGauge parses the value to string correctly based on the gauge type.